### PR TITLE
Remove 0.9.8 isms from the DSA and x509 verify bindings

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

**Comment by reaperhulk** [comment]
refs #2982

**Comment by lvh** [comment]
LGTM :)

File: src/_cffi_src/openssl/x509_vfy.py [hunk 1]
```diff
@@ -266,36 +266,9 @@
 static const long X509_V_FLAG_TRUSTED_FIRST = 0;
 #endif
 
-/* OpenSSL 1.0.0+ verification error codes */
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
 static const long Cryptography_HAS_100_VERIFICATION_ERROR_CODES = 1;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/x509_vfy.py]: I think we can probably remove this as it's unlikely that anybody used this as a feature check.

File: src/_cffi_src/openssl/x509_vfy.py [hunk 2]
```diff
@@ -266,36 +266,9 @@
...
-static const long X509_V_ERR_SUBTREE_MINMAX = 0;
-static const long X509_V_ERR_UNSUPPORTED_CONSTRAINT_TYPE = 0;
-static const long X509_V_ERR_UNSUPPORTED_CONSTRAINT_SYNTAX = 0;
-static const long X509_V_ERR_UNSUPPORTED_NAME_SYNTAX = 0;
-static const long X509_V_ERR_CRL_PATH_VALIDATION_ERROR = 0;
-#endif
 
-/* OpenSSL 1.0.0+ verification parameters */
-#if OPENSSL_VERSION_NUMBER >= 0x10000000L
 static const long Cryptography_HAS_100_VERIFICATION_PARAMS = 1;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/x509_vfy.py]: Same

File: src/_cffi_src/openssl/x509_vfy.py [hunk 3]
```diff
@@ -266,36 +266,9 @@
...
-#else
-static const long Cryptography_HAS_100_VERIFICATION_PARAMS = 0;
-static const long X509_V_FLAG_EXTENDED_CRL_SUPPORT = 0;
-static const long X509_V_FLAG_USE_DELTAS = 0;
-#endif
 
-/* OpenSSL 0.9.8recent+ */
-#ifdef X509_V_FLAG_CHECK_SS_SIGNATURE
 static const long Cryptography_HAS_X509_V_FLAG_CHECK_SS_SIGNATURE = 1;
```
Comment (reaperhulk) [inline src/_cffi_src/openssl/x509_vfy.py]: Same

**Comment by alex** [comment]
@stevepeak codecov appears to be stuck here.

**Pull request merged**